ChangeLog.d/fix-secure-element-key-creation.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix the error handling in psa_start_key_creation so that
+     out of memory issues are properly handled. Fixes #8537.
```

Comment:
* The changelog entry should mention the fact that this is only a problem when a non-default option is enabled. 
* `psa_start_key_creation` is an internal function, so it is not meaningful in a changelog entry. Changelog entries are meant for library users.

Suggested wording:

> Fix error handling when creating a key in a dynamic secure element (feature enabled by MBEDTLS_PSA_CRYPTO_SE_C). In a low memory condition, the creation could return PSA_SUCCESS but using or destroying the key would not work. Fixes #8537.